jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2720,9 +2720,17 @@ public String toLowerCase(Locale locale) {
 		}
 
 		if (StrHWAvailable() && language == "en") { //$NON-NLS-1$
-			String output = new String(lengthInternal());
-			if (toLowerHWOptimized(output))
-				return output;
+			int sLength = lengthInternal();
+
+			if (enableCompression && (null == compressionFlag || coder == LATIN1)) {
```

Comment:
We need to refactor this code and avoid the duplicate if. Can we merge with the path below?